routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+		_AddInternalServerError(ww, "CountUserAssociations: problem encoding response as JSON")
+		return
+	}
+}
+
+func (fes *APIServer) _constructUserAssociationQueryFromParams(
+	utxoView *lib.UtxoView, requestBody io.ReadCloser, queryType AssociationQueryType,
+) (*lib.UserAssociationQuery, error) {
+	var err error
+
+	// Decode request body.
+	decoder := json.NewDecoder(io.LimitReader(requestBody, MaxRequestBodySizeBytes))
+	requestData := UserAssociationQuery{}
+	if err = decoder.Decode(&requestData); err != nil {
+		return nil, errors.New("problem parsing request body")
+	}
+
+	// Parse TransactorPKID from TransactorPublicKeyBase58Check.
+	var transactorPKID *lib.PKID
+	if requestData.TransactorPublicKeyBase58Check != "" {
+		transactorPKID, err = fes.getPKIDFromPublicKeyBase58Check(
+			utxoView, requestData.TransactorPublicKeyBase58Check,
+		)
+		if err != nil {
+			return nil, errors.New("problem getting PKID for the transactor")
+		}
+	}
+
+	// Parse TargetUserPKID from TargetUserPublicKeyBase58Check.
+	var targetUserPKID *lib.PKID
+	if requestData.TargetUserPublicKeyBase58Check != "" {
+		targetUserPKID, err = fes.getPKIDFromPublicKeyBase58Check(
+			utxoView, requestData.TargetUserPublicKeyBase58Check,
+		)
+		if err != nil {
+			return nil, errors.New("problem getting PKID for the target user")
+		}
+	}
+
+	// Parse AppPKID from AppPublicKeyBase58Check
+	var appPKID *lib.PKID
+	if requestData.AppPublicKeyBase58Check != "" {
+		appPKID, err = fes.getPKIDFromPublicKeyBase58Check(
+			utxoView, requestData.AppPublicKeyBase58Check,
+		)
+		if err != nil {
+			return nil, errors.New("problem getting PKID for the app")
+		}
+	}
+
+	// Parse Limit.
+	if queryType == AssociationQueryTypeQuery {
+		if requestData.Limit < 0 || requestData.Limit > MaxAssociationsPerQueryLimit {
+			return nil, errors.New("invalid Limit provided")
+		}
+		if requestData.Limit == 0 {
+			requestData.Limit = MaxAssociationsPerQueryLimit
+		}
+	} else if queryType == AssociationQueryTypeCount {
+		if requestData.Limit != 0 {
+			return nil, errors.New("unsupported Limit param for count operation")
+		}
+	} else {
+		return nil, errors.New("invalid query type") // This can never happen.
+	}
+
+	// Parse LastSeenAssociationID (BlockHash) from LastSeenAssociationIdHex (string).
+	var lastSeenAssociationID *lib.BlockHash
+	if queryType == AssociationQueryTypeQuery {
+		if requestData.LastSeenAssociationID != "" {
+			lastSeenAssociationIdBytes, err := hex.DecodeString(requestData.LastSeenAssociationID)
+			if err != nil {
+				return nil, errors.New("invalid LastSeenAssociationID provided")
+			}
+			lastSeenAssociationID = lib.NewBlockHash(lastSeenAssociationIdBytes)
+		}
+	} else if queryType == AssociationQueryTypeCount {
+		if requestData.LastSeenAssociationID != "" {
+			return nil, errors.New("unsupported LastSeenAssociationID param for count operation")
+		}
+	} else {
+		return nil, errors.New("invalid query type") // This can never happen.
+	}
```

Comment:
let's use a switch - case statement for these. Additionally, I think we should do this towards the top of the function as these validations are cheaper than getting PKIDs from public keys. We can also move the validation on sort descending up there as well.